language-support/java/codegen/src/it/java/com/daml/ListTest.java

```diff
@@ -101,18 +102,14 @@ void fromProtobufValue() {
   }
 
   @Test
-  void fromJsonMyListRecord() throws JsonLfDecoder.Error {
+  void roundtripJsonMyListRecord() throws JsonLfDecoder.Error {
     MyListRecord expected =
         new MyListRecord(
             Arrays.asList(1L, 2L),
             Collections.singletonList(Unit.getInstance()),
             Arrays.asList(new Node<Long>(17L), new Node<Long>(42L)));
 
-    assertEquals(
```

Comment:
I'm wondering whether it's worth keeping this assertions (the one way conversion) in addition to the round-trip. It's possible that the roundtrip works, but doesn't encode what you expect, is it not?

Reply:
The output of the encoding is already tested in [JsonLfEncodersTest.java](https://github.com/digital-asset/daml/pull/17624/files#diff-be212d8f0c8833b9f47e3d955ecb97a515c7599c561ef1aa29edfa1bd7d3d38d)
and the decoding in `JsonLfDecodersTest.java`